src/features/tags/components/TagSelector.tsx

```diff
@@ -159,26 +159,28 @@ export function TagSelector({
   selectedTags,
   onAddTag,
   onRemoveTag,
-
   isReadOnly,
 }: TagSelectorProps) {
   const { data, isLoading, isError } = useTagDimensions()
 
   const groups = useMemo(
     () => data
-      ?.map(({ key: dimensionLabel, definedValues }) => {
+      ?.map(({ key, definedValues, displayName: dimensionDisplayName }) => {
         return {
-          label: dimensionLabel,
-          options: definedValues
-            .map(({ id: valueId, value: valueLabel }) => ({
-              label: valueLabel,
+          label: (dimensionDisplayName ?? key),
```

Comment:
No need for parens here and below
```suggestion
          label: dimensionDisplayName ?? key,
```